etc/scripts/get-stack.sh

```diff
@@ -748,11 +748,71 @@ on_path() {
 # Check whether ~/.local/bin is on the PATH, and print a warning if not.
 check_home_local_bin_on_path() {
   if ! on_path "$HOME_LOCAL_BIN" ; then
-    #TODO: offer to add it for the user (pull requests welcome!)
     info "WARNING: '$HOME_LOCAL_BIN' is not on your PATH."
     info "    Stack will place the binaries it builds in '$HOME_LOCAL_BIN' so"
     info "    for best results, please add it to the beginning of PATH in your profile."
     info ""
+    # detect which profile file to use, then print a message about updating it
+    if [ -n "$BASH_VERSION" ]; then
+      if [ -f "$HOME/.bash_profile" ]; then
+        profile_file="$HOME/.bash_profile"
+      else
+        profile_file="$HOME/.bashrc"
+      fi
+    elif [ -n "$ZSH_VERSION" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -n "$FISH_VERSION" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -n "$PROFILE" ]; then
+      profile_file="$PROFILE"
+    elif [ -f "$HOME/.bash_profile" ]; then
+      profile_file="$HOME/.bash_profile"
+    elif [ -f "$HOME/.bashrc" ]; then
+      profile_file="$HOME/.bashrc"
+    elif [ -f "$HOME/.zshrc" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -f "$HOME/.config/fish/config.fish" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -f "$HOME/.profile" ]; then
+      profile_file="$HOME/.profile"
+    else
+      info "    (profile not found; please add it to your PATH manually)"
+    fi
+
+    # print a message about updating profile (if found)
+    if [ -n "$profile_file" ]; then
+      info "    You can do this by running the following command:"
+      info "    echo 'export PATH=\"$HOME_LOCAL_BIN:\$PATH\"' >> \"$profile_file\""
+      info "    (You may need to restart your shell for this to take effect.)"
+      info ""
+    
+      # prompt to update it on their behalf, unless QUIET is set.
+      if [ -z "$QUIET" ]; then
+        while true; do
+          # default to no.
+          yesno=n
+          info "Would you like this installer to add it to your PATH in '$profile_file'?"
+          info "    (this will be done by adding export PATH=\"$HOME_LOCAL_BIN:\$PATH\" to it)"
+          info "    (you may need to restart your shell for this to take effect)"
+          info "    (y/N)"
```

Comment:
I think `(y/N)` should be `(y/n)`, given line 810?

Reply:
afaik convention is to capitalize the default selection. if it were `(Y/n)` then `yes` would be the presumed default.

Reply:
On conventions, a search found this discussion: https://ux.stackexchange.com/questions/53640/default-values-for-text-based-confirmation-prompt. I am not a Linux user, which may explain why I had not encountered it.

However, one thing I am seeking to do generally with the Stack project is reduce the level of 'assumed knowledge', including knowledge of conventions. In that regard, I think what the GHCup project does with its prompts is informative. See the `sh` script at https://get-ghcup.haskell.org and `ask_bashrc()`. GHCup 'spells it out'. Its equivalent would be something like:

`[Y] Yes, prepend  [N] No (default is "N").`